be/src/vec/exprs/lambda_function/varray_match_function.cpp

```diff
@@ -0,0 +1,188 @@
+            const auto& column_array_nullmap =
+                    assert_cast<const ColumnNullable*>(second_column.get())->get_null_map_column();
+            VectorizedUtils::update_null_map(second_outside_null_map->get_data(),
+                                             column_array_nullmap.get_data());
+        }
+        const ColumnArray& second_col_array = assert_cast<const ColumnArray&>(*second_arg_column);
+        const auto& second_off_data = assert_cast<const ColumnArray::ColumnOffsets&>(
+                                              second_col_array.get_offsets_column())
+                                              .get_data();
+        const auto& second_nested_null_map_data =
+                assert_cast<const ColumnNullable&>(*second_col_array.get_data_ptr())
+                        .get_null_map_column()
+                        .get_data();
+        const auto& second_nested_column =
+                assert_cast<const ColumnNullable&>(*second_col_array.get_data_ptr())
+                        .get_nested_column();
+        const auto& second_nested_data =
+                assert_cast<const ColumnUInt8&>(second_nested_column).get_data();
+
+        //3. calculate result column with second bool column
+        for (int row = 0; row < input_rows; ++row) {
+            //first or second column is null, so current row is invalid data
+            if (first_outside_null_map->get_data()[row] ||
+                second_outside_null_map->get_data()[row]) {
+                result_null_column->get_data()[row] = 1;
+                result_data_column->get_data()[row] = 0;
+            } else {
+                // has_null in current array
+                bool has_null_elem = false;
+                // res for current array
+                bool res_for_array = MATCH_ALL;
+                auto first_offset_start = first_off_data[row - 1];
+                auto first_offset_end = first_off_data[row];
+                auto second_offset_start = second_off_data[row - 1];
+                auto second_offset_end = second_off_data[row];
+                auto move_off = second_offset_start;
+                for (auto off = first_offset_start;
+                     off < first_offset_end && move_off < second_offset_end; // not out range
+                     ++off) {
+                    if (first_nested_nullable_column.is_null_at(off)) {
+                        has_null_elem = true;
+                    } else {
+                        if (!second_nested_null_map_data[move_off] &&    // not null
+                            second_nested_data[move_off] != MATCH_ALL) { // not match
+                            res_for_array = !MATCH_ALL;
+                            break;
+                        } // default is MATCH_ALL
+                    }
+                    move_off++;
+                }
+                result_null_column->get_data()[row] = has_null_elem && res_for_array == MATCH_ALL;
```

Comment:
Is this semantic right, what about array_match_all(x ->x=1, array), will this resut null for  [1, null, 0]  ?

Reply:
return false

Reply:
add such case pls， result_null_column->get_data()[row]  will be true and lead to null